src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,847 @@
+
+@section ft_leaf_reconciliation Leaf (child) reconciliation
+
+Deleted pages are already on disk and inherently never themselves need to be reconciled.
+Instantiated pages that have been read into memory, however, do.
+
+Eviction of instantiated pages where the truncation is unresolved is blocked.
+In principle these pages could successfully go through update-restore eviction, but there
+are complications involved in doing so (e.g. handling the update list, and if the page
+were to split the \c page_del information would have to be cloned, and that requires
+locking) and it was judged not worthwhile.
+
+Eviction of instantiated pages where the truncation is committed is permitted, however,
+and checkpoint of instantiated pages is always allowed.
+The first reconciliation after instantiation clears the instantiated flag (\c
+page->modify->instantiated) and discards the \c page_del structure, as once a
+reconciliation result exists it can be used to reconcile the parent.
+(This is true whether or not it includes the tombstones from instantiation, i.e., whether
+the truncation was committed.)
+
+Note that "unresolved" includes "prepared".
+While we cannot write out a prepared truncation in the parent page's address cells, in
+principle after instantiation we could write out the prepared tombstone updates like any
+other prepared updates, and after doing so nothing special is needed in the parent.
+This is not currently done, chiefly because safely determining whether the truncation is
+prepared at the point where eviction needs to check is problematic.
+(We can't check \c page_del because it might have been discarded by then; looking in \c
+inst_updates is at best messy and it isn't entirely clear what locking or synchronization
+might be needed.
+Some such check, or an extra flag in the modify structure, is probably possible if it
+eventually becomes important to allow these evictions.)
+
+@section ft_on_disk_format On-disk format
+
+Deleted pages are referred to on disk by a special address cell type, \c WT_CELL_ADDR_DEL.
+These contain three additional packed integers between the time aggregate and the cell
+data: the transaction ID, commit timestamp, and durable timestamp of the transaction that
+deleted the page.
+(Only committed truncations are written out.
+Prepared truncations cannot be represented on disk.
+Truncations that are globally visible do not result in a cell in the parent page at all.)
+
+These fields are, however, only present if the page header includes the \c
+WT_PAGE_FT_UPDATE flag, whose value is 0x20.
+Proper support for timestamped fast-truncate only appeared in WT 11.0; earlier versions
+neither write these fields nor expect to find them.
+The explicit header flag is required to make compatibility guarantees function as needed.
+(MongoDB 6.x knows how to read pages with \c WT_PAGE_FT_UPDATE set, but does not write
+them.
+This is controlled in the WT library by the build config.)
```

Comment:
@sauclovian-wt, can you elaborate on this? I am not sure to understand. What building parameter(s) need(s) to be changed?

Reply:
Currently it's controlled by WT_STANDALONE_BUILD, indirectly by a flag in __wt_process. I didn't want to be too explicit because it might change, but I'll fix it to at least name the flag. (In [af26591](https://github.com/wiredtiger/wiredtiger/pull/8237/commits/af265919038ffe7086fbdc1db36c6800c63bc466))